sdk/provisioning/Azure.Provisioning/src/ProvisioningDeployment.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Provisioning.Primitives;
+using Azure.ResourceManager.Resources;
+
+// This needs to go, but we should discuss whether to directly translate or
+// we should just simplify it here and let you drop down to ARM if you need more.
+using ArmProvisioningState = Azure.ResourceManager.Resources.Models.ResourcesProvisioningState;
+
+namespace Azure.Provisioning;
+
+// This is a draft to explore convenience APIs to make it easier to deploy
+// and immediately make use of Azure resources.  It's likely going to change
+// a fair bit in the near future.
+
+/// <summary>
+/// Represents the result of a live deployment including any errors and outputs.
+/// </summary>
+public class ProvisioningDeployment
+{
+    /// <summary>
+    /// Gets the provisioning context that was used to deploy the resources.
+    /// </summary>
+    internal ProvisioningContext Context { get; }
+
+    /// <summary>
+    /// Gets the <see cref="ArmDeploymentResource"/> that was used to deploy
+    /// the infrastructure template.
+    /// </summary>
+    public ArmDeploymentResource Deployment { get; }
+
+    /// <summary>
+    /// Gets the <see cref="ArmProvisioningState"/> of the deployment.
+    /// </summary>
+    public ArmProvisioningState? ProvisioningState => Deployment.Data.Properties.ProvisioningState;
+
+    /// <summary>
+    /// Gets any optional errors that occurred during the deployment.
+    /// </summary>
+    public ResponseError? Error => Deployment.Data.Properties.Error;
+
+    /// <summary>
+    /// Gets a dictionary of template outputs from the deployment.  This is a
+    /// simple name/value mapping.  You can also get the values directly from
+    /// any <see cref="BicepOutput"/> instances as they'll have been updated
+    /// after a successful deployment and carry additional information - like
+    /// whether the value is considered secure.
+    /// </summary>
+    public IReadOnlyDictionary<string, object?> Outputs { get; }
```

Comment:
We can do similarly neat things if you deploy live resources like give you easy access to all the `BicepOutput`s.